Source/WebCore/dom/ProcessingInstruction.cpp

```diff
@@ -122,55 +123,42 @@ void ProcessingInstruction::checkStyleSheet()
                 URL finalURL({ }, m_localHref);
                 m_sheet = XSLStyleSheet::createEmbedded(*this, finalURL);
                 m_loading = false;
-                document().scheduleToApplyXSLTransforms();
+                document->scheduleToApplyXSLTransforms();
             }
 #endif
         } else {
-            if (m_cachedSheet) {
-                m_cachedSheet->removeClient(*this);
-                m_cachedSheet = nullptr;
-            }
-
-            if (m_loading) {
-                m_loading = false;
-                document().styleScope().removePendingSheet(*this);
-            }
-
-            Ref<Document> originalDocument = document();
-
-            String url = document().completeURL(href).string();
-
-            bool didEventListenerDisconnectThisElement = !isConnected() || &document() != originalDocument.ptr();
-            if (didEventListenerDisconnectThisElement)
```

Comment:
Why are we dropping this early return?

Reply:
This code used to fire a beforeload event as you can see in https://commits.webkit.org/187003@main.
 That event was dropped but the code wasn't cleaned up. Since no JS runs, anymore, there is no need to check if we got disconnected anymore.

Reply:
Please elaborate that in the commit message.